### PR TITLE
HMAN-166 ApiResponse messages should use ValidationError Class

### PR DESCRIPTION
**Pull request by marvincudjoe**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/HMAN-166

### Change description ###
@APIResponse uses variables from the ValidationError class.
Added a few more variables for the relevant responses 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/hmc/controllers/HearingActualsManagementController.java [hunk 1]
```diff
@@ -28,14 +29,16 @@ public HearingActualsManagementController(HearingActualsService hearingActualsSe
...
-        @ApiResponse(code = 500, message = "Error occurred on the server")
+        @ApiResponse(code = 200, message = ValidationError.HEARING_ACTUALS_SUCCESSFULLY_PROCESSED),
+        @ApiResponse(code = 400,
+            message = "One or more of the following reasons:"
+            + "\n1) " + ValidationError.HEARING_ACTUALS_NO_HEARING_RESPONSE_FOUND
+            + "\n2) " + ValidationError.INVALID_HEARING_ID_DETAILS
+            + "\n001) " + ValidationError.HEARING_ACTUALS_ID_NOT_FOUND
+            + "\n002) " + ValidationError.HEARING_ACTUALS_INVALID_STATUS
+            + "\n003) " + ValidationError.HEARING_ACTUALS_HEARING_DAYS_INVALID
+            + "\n004) " + ValidationError.HEARING_ACTUALS_NON_UNIQUE_HEARING_DAYS),
```
Comment (RebeccaBaker) [inline src/main/java/uk/gov/hmcts/reform/hmc/controllers/HearingActualsManagementController.java:40]: the numbering seems to be incorrect

File: src/main/java/uk/gov/hmcts/reform/hmc/controllers/LinkHearingGroupController.java [hunk 2]
```diff
@@ -34,12 +35,13 @@ public LinkHearingGroupController(LinkedHearingGroupService linkedHearingGroupSe
...
-        @ApiResponse(code = 400, message = "003 hearing request already in a group"),
-        @ApiResponse(code = 400, message = "004 invalid state of hearing for request"),
-        @ApiResponse(code = 400, message = "005 Hearing Order is not unique")
+        @ApiResponse(code = 400,
+            message = "One or more of the following reasons:"
+                + "\n001) " + ValidationError.HEARINGS_IN_GROUP_SIZE
+                + "\n002) " + ValidationError.HEARING_REQUEST_CANNOT_BE_LINKED
+                + "\n003) " + ValidationError.HEARING_REQUEST_ALREADY_LINKED
+                + "\n004) " + ValidationError.INVALID_STATE_FOR_HEARING_REQUEST
+                + "\n005) " + ValidationError.HEARING_ORDER_NOT_UNIQUE)
```
Comment (RebeccaBaker) [inline src/main/java/uk/gov/hmcts/reform/hmc/controllers/LinkHearingGroupController.java:44]: can we change numbering to 1,2,3,4,5 not 001 etc

File: src/main/java/uk/gov/hmcts/reform/hmc/controllers/LinkHearingGroupController.java [hunk 3]
```diff
@@ -49,14 +51,16 @@ public HearingLinkGroupResponse validateLinkHearing(@RequestBody @Valid
...
+        @ApiResponse(code = 200, message = "Success"),
+        @ApiResponse(code = 400,
+            message = "One or more of the following reasons:"
+                + "\n1) " + ValidationError.INVALID_LINKED_GROUP_REQUEST_ID_DETAILS
+                + "\n001) " + ValidationError.HEARINGS_IN_GROUP_SIZE
+                + "\n002) " + ValidationError.HEARING_REQUEST_CANNOT_BE_LINKED
+                + "\n003) " + ValidationError.HEARING_REQUEST_ALREADY_LINKED
+                + "\n004) " + ValidationError.INVALID_STATE_FOR_HEARING_REQUEST
+                + "\n007) " + ValidationError.INVALID_STATE_FOR_LINKED_GROUP
+                + "\n008) " + ValidationError.INVALID_STATE_FOR_UNLINKING_HEARING_REQUEST)
```
Comment (RebeccaBaker) [inline src/main/java/uk/gov/hmcts/reform/hmc/controllers/LinkHearingGroupController.java:63]: same again

File: src/main/java/uk/gov/hmcts/reform/hmc/controllers/PartiesNotifiedController.java [hunk 4]
```diff
@@ -33,10 +34,13 @@ public PartiesNotifiedController(PartiesNotifiedService partiesNotifiedService)
...
-        @ApiResponse(code = 404, message = "001 no such id"),
-        @ApiResponse(code = 404, message = "002 no such response version"),
-        @ApiResponse(code = 500, message = "Error occurred on the server")
+        @ApiResponse(code = 400, message = ValidationError.INVALID_HEARING_ID_DETAILS),
+        @ApiResponse(code = 404,
+            message = "One or more of the following reasons:"
+                + "\n001) " + ValidationError.PARTIES_NOTIFIED_ID_NOT_FOUND
+                + "\n002) " + ValidationError.PARTIES_NOTIFIED_RESPONSE_VERSION_MISMATCH
+        ),
+        @ApiResponse(code = 500, message = ValidationError.INTERNAL_SERVER_ERROR)
```
Comment (RebeccaBaker) [inline src/main/java/uk/gov/hmcts/reform/hmc/controllers/PartiesNotifiedController.java:43]: and here

File: src/main/java/uk/gov/hmcts/reform/hmc/controllers/PartiesNotifiedController.java [hunk 5]
```diff
@@ -47,9 +51,15 @@ public void putPartiesNotified(@RequestBody @Valid PartiesNotified partiesNotifi
...
-        @ApiResponse(code = 404, message = "Hearing id not found")
+        @ApiResponse(code = 200, message = ValidationError.HEARING_ID_IS_VALID),
+        @ApiResponse(code = 400,
+            message = "One or more of the following reasons:"
+                + "\n1) " + ValidationError.INVALID_HEARING_ID_DETAILS
+                + "\n003) " + ValidationError.PARTIES_NOTIFIED_ALREADY_SET),
+        @ApiResponse(code = 404,
+            message = "One or more of the following reasons:"
+                + "\n001) " + ValidationError.PARTIES_NOTIFIED_ID_NOT_FOUND
+                + "\n002) " + ValidationError.PARTIES_NOTIFIED_RESPONSE_VERSION_MISMATCH)
```
Comment (RebeccaBaker) [inline src/main/java/uk/gov/hmcts/reform/hmc/controllers/PartiesNotifiedController.java:62]: plus this one

File: src/main/java/uk/gov/hmcts/reform/hmc/controllers/HearingActualsManagementController.java [hunk 6]
```diff
@@ -28,14 +29,16 @@ public HearingActualsManagementController(HearingActualsService hearingActualsSe
...
     @ApiResponses(value = {
-        @ApiResponse(code = 200, message = "Hearing actuals update processed"),
-        @ApiResponse(code = 400, message = "Invalid hearing Id"),
-        @ApiResponse(code = 400, message = "001 No such id: hearingId"),
-        @ApiResponse(code = 400, message = "002 invalid status HEARING_REQUESTED"),
-        @ApiResponse(code = 400, message = "002 invalid status AWAITING_LISTING"),
-        @ApiResponse(code = 400, message = "003 invalid date"),
-        @ApiResponse(code = 400, message = "004 non-unique dates"),
-        @ApiResponse(code = 500, message = "Error occurred on the server")
+        @ApiResponse(code = 200, message = ValidationError.HEARING_ACTUALS_SUCCESSFULLY_PROCESSED),
```
Comment (RebeccaBaker) [inline src/main/java/uk/gov/hmcts/reform/hmc/controllers/HearingActualsManagementController.java:32]: for all successful flows can they be reverted back to a string and removed from Validation.Error

**Review by RebeccaBaker — CHANGES_REQUESTED** [review]

**Review by padmapriyanalam — APPROVED** [review]

**Pull request merged**